src/Savegame/BattleUnit.cpp

```diff
@@ -292,7 +294,20 @@ void BattleUnit::prepareUnitSounds()
 			_deathSound = _armor->getFemaleDeathSounds();
 	}
 }
-
+/**
+ * Helper function preparing tag names.
+ */
+void BattleUnit::prepareTagNames(const ScriptGlobal *shared)
+{
+	_tagData.clear();
+	
+	ArgEnum index = ScriptParserBase::getArgType<ScriptTag<BattleUnit>>();
+	
+	for (size_t i = 0; i < shared->getTagNames().at(index).values.size(); ++i)
```

Comment:
could you use `for (const auto& v : shared->getTagNames().at(index).values)`? It look better and work better (less memory visits and operations per iteration)

Reply:
Sure, will do